### PR TITLE
Fix runtime errors reported when using long input sequence lengths with LoRA

### PR DESCRIPTION
**Pull request by vivekgoe**
This PR has following fixes,
-  Increase size of indices tensors used to maintain multi-lora state information from max_num_batched_tokens to 3*max_num_batched_tokens. This increase is done to provide buffer for padding done in batch & sequence dimensions.
-  Move logic to remove padding from lora_logits from execute_model() back to Class LogitsProcessorWithLoRA, this is done to fix race condition caused by updating multi-lora state information directly.   

FIX https://github.com/HabanaAI/vllm-fork/issues/237

### CODE REVIEW

File: vllm/worker/habana_model_runner.py [hunk 1]
```diff
@@ -1350,9 +1350,9 @@ def warmup_scenario(self,
         times = 3 if use_graphs or is_pt_profiler_run else 1
         if self.lora_config and not is_lora_profile_run:
             lora_mapping = LoRAMapping(
-                [0] * batch_size * seq_len,
-                [0] * batch_size * seq_len,
-            )
+                **dict(index_mapping=[0] * batch_size * seq_len,
```
Comment (michalkuligowski) [inline vllm/worker/habana_model_runner.py:1353]: how does introducing dict here affect memory footprint? since we dont have control on when its garbage collected and size grows with batch_size*seq_len

Comment (vivekgoe) [reply vllm/worker/habana_model_runner.py:1353]: @michalkuligowski Nothing has changed from existing code, we are still populating the same dataclass (LoRAMapping) as before,  **dict simply is to use key, value notation initialization of class members.  Regarding memory footprint, this dataclass needs to be available all the time if LoRA is enabled, it is a must have for LoRA operation. Again nothing specific to Gaudi/HPU, this is the way LoRA feature is implemented in vLLM.

**Review by hlahkar — APPROVED** [review]
